src/Coordination/KeeperStateMachine.cpp

```diff
@@ -862,12 +862,16 @@ uint64_t KeeperStateMachine::getKeyArenaSize() const
     return storage->getArenaDataSize();
 }
 
-uint64_t KeeperStateMachine::getLatestSnapshotBufSize() const
+uint64_t KeeperStateMachine::getLatestSnapshotSize() const
 {
     std::lock_guard lock(snapshots_lock);
-    if (latest_snapshot_buf)
-        return latest_snapshot_buf->size();
-    return 0;
+    if (latest_snapshot_info.disk == nullptr)
+        return 0;
+
+    if (!latest_snapshot_info.size.has_value())
+        latest_snapshot_info.size = latest_snapshot_info.disk->getFileSize(latest_snapshot_info.path);
```

Comment:
This call to `getFileSize()` will be done under `snapshots_lock`, and it could occasionally slow down others who grab this lock.
No sure if this a big deal. WDYT?

Reply:
Yeah, I wouldn't say it's a big deal because the lock is taken only for reading/writing/applying snapshot which are all much heavier operations than fetching size.
Also they are not done that often + the size is cached.

Reply:
But I can minimize it by storing `latest_snapshot_info` inside a `std::shared_ptr` and making size `atomic`.
It's okay if we return a size for a snapshot that become old 1ms before.

Reply:
okay this escalated a bit, turned it mini refactor...